app/src/main/java/org/covidwatch/android/CovidWatchApplication.kt

```diff
@@ -44,4 +48,23 @@ class CovidWatchApplication : Application() {
             downloadRequest
         )
     }
+
+    fun executePublicContactEventsRefresh() {
+        val downloadRequest = OneTimeWorkRequestBuilder<ContactEventsDownloadWorker>()
+            .setConstraints(getRefreshConstraints())
+            .build()
+
+        WorkManager.getInstance(this).enqueueUniqueWork(
+            ContactEventsDownloadWorker.WORKER_NAME,
+            ExistingWorkPolicy.REPLACE,
+            downloadRequest
+        )
+    }
```

Comment:
Lets keep Application class clean, and move this work to a separate class?
```kotlin
class UpdateContactEventsUseCase(private val workManager: WorkManager) {
  fun execute() {
    workManager.enqueueUniqueWork()
  }
}
```
We can then declare it in `AppModule.kt`, and provide it to HomeViewModel via constructor

Reply:
We already have the code for scheduling periodic refresh there. I think it's good to have it in a single place. Or do you mean to extract all WorkManager-related code?

Reply:
Let's keep PeriodicWorkRequestBuilder here, but move out OneTimeWorkRequestBuilder.

I'm writing this because we shouldn't be accessing Application class from ViewModel:
~~`CovidWatchApplication.getContext()?.executePublicContactEventsRefresh()`~~

Here is why
>Ideally, ViewModels shouldn’t know anything about Android. This improves testability, leak safety and modularity. A general rule of thumb is to make sure there are no android.* imports in your ViewModels (with exceptions like android.arch.*). The same applies to presenters.

https://medium.com/androiddevelopers/viewmodels-and-livedata-patterns-antipatterns-21efaef74a54

Reply:
All right. But I don't like a solution where we split similar code in different places so I'd vote to move all WorkManager code in another class.

Reply:
I think you're right about keeping these Workers in the same place. **Not in this PR**, but later we can create some `Enqueuer` class that will have two methods, one for PeriodicWorkRequestBuilder and the other for OneTimeWorkRequestBuilder

Reply:
Added a commit for this.

Reply:
I agree with moving our the logic from Application class! Also, I think it's better to do that now. 

Let's move these `schedulePeriodicPublicContactEventsRefresh` and `executePublicContactEventsRefresh` methods into separate `UseCase` classes such as `NotifyAboutPossibleExposureUseCase`. It will make this logis as isolated as possible so we could reuse it anywhere in the app. It also should make it testable as we provide `WorkManager` as a constructor parameter and can mock it.

Reply:
Well, `ContactEventsDownloader` works too.